docs/reference/yaml.md

```diff
@@ -239,7 +239,60 @@ The meanings and formats of `limits` and `requests` may vary depending on whethe
 
 See docs for [Docker Swarm](https://docs.docker.com/config/containers/resource_constraints/) or for [Kubernetes](https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/#how-pods-with-r    esource-limits-are-run).
 
-### YAML - environment variable substitution
+## Configuration
+The configuration section allows you to define additional configuration that is global to the entire stack, currently this mostly impacts function build time options.
+
+### Templates
+The `templates` list allows you to define the information required to pull the templates for your functions.  This list of templates will automatically be pulled when you build your functions. When configured correctly, this allows you to completely build your functions with just `faas-cli build`.  Without this section, you must manually `faas-cli template pull <source>` _before_ you use `faas-cli build`.
+
+```yaml
+configuration:
+  templates:
+    - name: python3-http
+      source: https://github.com/openfaas-incubator/python-flask-template
+```
+
+### Copy
+The `copy` list allows you to define additional project paths that will be copied into your function's handler folder.
+
+```yaml
+configuration:
+  copy:
+    - ./common
+    - ./data
+    - ./models
+```
+
+Given the above configuration, the build folder for a python `hello` function would look like
+
+```sh
```

Comment:
I'd clean up this tree output to be like the basic output only, with the `copy` folders to make it more clear what was being added. Here, you have `data_set_n`, `utils.py`, etc that might be a little confusing. I had to look through the list twice to notice the copy folders.
I see this is a real example, but I think it would be clearer if it was instead just the default new function with the copy folders

Reply:
Hi @burtonr sorry it took so long to respond, i was busy for a day and then i got sick for a bit. 

For this, do you think this is better?

```sh
build
└── hello
    ├── Dockerfile
    ├── function
    │   ├── __init__.py
    │   ├── models
    │   │   └── ...
    │   ├── data
    │   │   └── ...
    │   ├── common
    │   │   └── ...
    │   ├── handler.py
    │   └── requirements.txt
    ├── index.py
    ├── requirements.txt
    └── template.yml
```

Reply:
No worries. That looks clearer that the 3 directories being copied are included and where they are, although I don't remember what it looked like before... this does look good!